wagtail/contrib/forms/models.py

```diff
@@ -291,25 +289,33 @@ def serve_preview(self, request, mode_name):
         if mode_name == "landing":
             return self.render_landing_page(request)
         else:
-            return super().serve_preview(request, mode_name)
+            return self.serve_preview(request, mode_name)
 
 
 def validate_to_address(value):
     for address in value.split(","):
         validate_email(address.strip())
 
 
-class AbstractEmailForm(AbstractForm):
+class AbstractForm(FormPageMixin, Page):
+    """A Form Page. Pages implementing a form should inherit from it."""
+
+    class Meta:
+        abstract = True
+
+
+class AbstractEmailForm(FormPageMixin, Page):
     """
```

Comment:
Should we also pull out the email from page to a mixin?

Reply:
> Thanks again. Picking up an old PR is tricky but super keen to see this in as it will make working with this form builder module more flexible.
> 
> Really appreciate the time spent on this @mehrdadmoradii

@lb- Thank you for the feedback!

Reply:
> Should we also pull out the email from page to a mixin?

@lb- I think that is also a good idea, so we can be more consistent with our API. Should I open a new issue for that problem?

Reply:
i think it would be good if this PR can be updated to also have email form as a mixin. Read the original PR's comments for some notes about this.

Reply:
> i think it would be good if this PR can be updated to also have email form as a mixin. Read the original PR's comments for some notes about this.

Got it!